nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -25,7 +24,7 @@
 my $dryRestartByActivationFile = "/run/nixos/dry-activation-restart-list";
 my $dryReloadByActivationFile = "/run/nixos/dry-activation-reload-list";
 
-make_path("/run/nixos", { mode => 0755 });
+make_path("/run/nixos", { mode => oct(755) });
```

Comment:
https://perldoc.perl.org/File::Path - For some reason the official documentation also doesn't use the oct-function but it should be fine though. I assume the Perl function behaves like `chmod` regarding the missing `0` ("Omitted digits are assumed to be leading zeros.").